packages/frontpage/app/(app)/layout.tsx

```diff
@@ -86,15 +97,42 @@ export default async function Layout({
 async function LoginOrLogout() {
   const session = await getSession();
   if (session) {
+    const did = await getDidFromHandleOrDid(session.user.name as string);
     return (
-      <form
-        action={async () => {
-          "use server";
-          await signOut();
-        }}
-      >
-        <button type="submit">Logout ({session.user.name})</button>
-      </form>
+      <DropdownMenu>
+        <DropdownMenuTrigger>
+          {did ? (
+            <UserAvatar did={did} size="smedium" />
+          ) : (
+            <span>{session.user.name}</span>
+          )}
+        </DropdownMenuTrigger>
+        <DropdownMenuContent className="w-56" side="bottom" align="end">
+          <form
+            action={async () => {
+              "use server";
+              await signOut();
+            }}
+          >
+            <DropdownMenuLabel>My Account</DropdownMenuLabel>
+            <DropdownMenuSeparator />
+            <DropdownMenuItem asChild>
+              <Link
+                href={`/profile/${session.user.name}`}
+                className="cursor-pointer"
+              >
+                Profile
+              </Link>
+            </DropdownMenuItem>
+            <DropdownMenuSeparator />
+            <DropdownMenuItem>
```

Comment:
```suggestion
            <DropdownMenuItem asChild>
```

I think you want `asChild` here. Currently there are nested interacted elements which is probs not accessible. Regardless, there's an extra dom element that doesn't need to be there.